src/notifications/mod.rs

```diff
@@ -0,0 +1,310 @@
+            json!(map),
+        )
+    }
+
+    /// Delete the thread subscription.
+    ///
+    /// See the [github docs](https://developer.github.com/v3/activity/notifications/#delete-a-thread-subscription)
+    /// for more information.
+    pub fn delete_subscription<S>(&self, id: S) -> Future<()>
+    where
+        S: Into<String>,
+    {
+        self.github.delete(&format!(
+            "/notifications/threads/{}/subscription",
+            id.into()
+        ))
+    }
+}
+
+// representations
+
+#[derive(Debug, Deserialize)]
+pub struct Thread {
+    pub id: String,
+    pub unread: bool,
+    pub updated_at: String,
+    pub last_read_at: Option<String>,
+    pub reason: String,
+    pub subject: Subject,
+    pub repository: Repository,
+    pub url: String,
+    pub subscription_url: String,
+}
+
+#[derive(Default)]
+pub struct ThreadListOptions {
+    params: HashMap<&'static str, String>,
+}
+
+impl ThreadListOptions {
+    pub fn builder() -> ThreadListOptionsBuilder {
+        ThreadListOptionsBuilder::new()
+    }
+
+    /// serialize options as a string. returns None if no options are defined
+    pub fn serialize(&self) -> Option<String> {
+        if self.params.is_empty() {
+            None
+        } else {
+            let encoded: String = form_urlencoded::Serializer::new(String::new())
+                .extend_pairs(&self.params)
+                .finish();
+            Some(encoded)
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct ThreadListOptionsBuilder(ThreadListOptions);
+
+impl ThreadListOptionsBuilder {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// if `true`, show notifications marked as read. Default: `false`
+    pub fn all(&mut self, all: bool) -> &mut Self {
+        self.0.params.insert("all", all.to_string());
+        self
+    }
+
+    /// if `true`, only shows notifications in which the user is directly participating or
+    /// mentioned. Default: `false`
+    pub fn participating(&mut self, val: bool) -> &mut Self {
+        self.0.params.insert("participating", val.to_string());
+        self
+    }
+
+    /// Only show notifications updated after the given time.
+    pub fn since<T>(&mut self, since: T) -> &mut Self
+    where
+        T: Into<String>,
+    {
+        self.0.params.insert("since", since.into());
+        self
+    }
+
+    /// Only show notifications updated before a given time.
+    pub fn before<T>(&mut self, before: T) -> &mut Self
+    where
+        T: Into<String>,
+    {
+        self.0.params.insert("before", before.into());
+        self
+    }
+
+    pub fn build(&self) -> ThreadListOptions {
+        ThreadListOptions {
+            params: self.0.params.clone(),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Subject {
+    title: String,
+    url: String,
+    latest_comment_url: String,
+    #[serde(rename = "type")]
+    kind: String,
+}
+
+#[derive(Debug, Deserialize)]
```

Comment:
I've created a new smaller `Repository` type because for some reason the github api returns here a repository object with less properties. i had to make like 5 fields optional to make to work with `hubcaps::repositories::Repository`.